src/components/SharedComponents/PermissionGateContainer.js

```diff
@@ -0,0 +1,160 @@
+// @flow
+
+import { useNavigation } from "@react-navigation/native";
+import Modal from "components/SharedComponents/Modal";
```

Comment:
These PermissionGates might actually be good candidates for using react-navigation modals, since they're full screen and don't need background opacity, etc.